src/psyclone/psyad/transformations/preprocess.py

```diff
@@ -54,8 +55,14 @@ def preprocess_trans(kernel_psyir):
 
     '''
     dot_product_trans = DotProduct2CodeTrans()
+    matmul_trans = Matmul2CodeTrans()
 
     for oper in kernel_psyir.walk(BinaryOperation):
         if oper.operator == BinaryOperation.Operator.DOT_PRODUCT:
             # Apply DOT_PRODUCT transformation
             dot_product_trans.apply(oper)
+
+    for oper in kernel_psyir.walk(BinaryOperation):
```

Comment:
You could do away with this separate loop (and thus its associated `walk`) and just have the `if` as a second child of the loop at line 60.

Reply:
As discussed privately I was concerned about the transformations affecting each other. However this is not the case, so I have done as you suggest. When we add support nested intrinsics (see #1561 and #1562 for example) we might need to revisit - and perhaps even have an outer `while intrinsics` loop. However, the tests associated with implementing solutions to those issues should pick up any problems.